pkg/processor/sourceprocessor/source_processor.go

```diff
@@ -239,6 +249,29 @@ func (sp *sourceProcessor) ProcessLogs(ctx context.Context, md pdata.Logs) (pdat
 		if sp.isFilteredOut(atts) {
 			rs.InstrumentationLibraryLogs().RemoveIf(func(pdata.InstrumentationLibraryLogs) bool { return true })
 		}
+
+		// Set stream and time attributes and log body for docker log
+		ills := rs.InstrumentationLibraryLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ill := ills.At(j)
+			logs := ill.Logs()
+			for k := 0; k < logs.Len(); k++ {
+				log := logs.At(k)
+				if log.Body().Type() == pdata.AttributeValueTypeString {
+					err := json.Unmarshal([]byte(log.Body().StringVal()), &dockerLog)
+
+					// Skip if error occures or any of the expected fields is empty
+					if err != nil || dockerLog.Stream == "" || dockerLog.Time == "" || dockerLog.Log == "" {
```

Comment:
Do we not want to log any errors, warnings?

Reply:
As customer can have CRI-O logs I don't think so
eventually, we can log everything and add configuration option to enable/disable docker logs parsing

Reply:
Oh I see, so it's not an erroneous situation when an error is returned from `json.Unmarshal`, it just means the log does not come from Docker runtime.